composables/useToggleApprovall.js

```diff
@@ -1,33 +1,49 @@
 import { useMutation, useQueryClient } from "@tanstack/vue-query";
 import { createToast } from "mosha-vue-toastify";
 import { useSingleReport } from "./useSingleReport";
+import axios from "axios";
 
-export const useToggleApproval = () => {
+export const useToggleApproval = ({ eula }) => {
   const queryClient = useQueryClient();
   const { report } = useSingleReport();
   const route = useRoute();
-  const { mutateAsync: mutateApproval } = useMutation({
-    mutationFn: ({ pkg, eula }) =>
-      $fetch(`/api/reports/${route.params.id}`, {
-        method: "PUT",
-        body: {
-          ...report.value.state,
-          [pkg]: {
-            ...report.value?.state?.[pkg],
-            [eula]: report.value?.state?.[pkg]?.[eula] ? false : true,
-          },
-        },
-      }),
-    onSuccess: () =>
-      queryClient.invalidateQueries({ queryKey: ["report", route.params.id] }),
-    onError: () =>
-      createToast("There has been an issue toggling approval", {
-        type: "danger",
-        hideProgressBar: true,
-      }),
+
+  const createBody = (value) => ({
+    ...report.value.state,
+    [route.query.name]: {
```

Comment:
Changed back but I wonder what you think can go wrong?

Reply:
Generally, I prefer directly passing arguments to taking them off globals that may or may not have a direct relationship because in the former case, it is clear when changes are made that they affect this particular thing. In the case of the latter, somone could change the way names are rendered in the query for other reasons and not realize its going to affect how the data is stored or even make it impossible to look it up. Sometimes there are other trade-offs, but in this case it is pretty simple to be direct and that way reduce a possible bug vector.

(Also, I don't see the change reflected in the diffs. Is it possible there are some un-pushed local changes?)